src/types.ts

```diff
@@ -29,20 +29,31 @@ export type SchemaType =
   | "[]int"
   | "[]uint"
   | "[]uuid";
-// TODO: index signature is a better fit here imo.
-// also look into eslint config to allow for usage of index signatures
-export type Schema = Record<
-  string,
-  {
-    type?: SchemaType;
-    filterable?: boolean;
-    /**
-     * @deprecated use `full_text_search` instead
-     */
-    bm25?: boolean | Partial<FTSParams>;
-    full_text_search?: boolean | Partial<FTSParams>;
-  }
->;
+
+export type Schema =
+  | {
+      [K in string]: {
+        type?: SchemaType;
+        filterable?: boolean;
+        /**
+         * @deprecated use `full_text_search` instead
+         */
+        bm25?: boolean | Partial<FTSParams>;
+        full_text_search?: boolean | Partial<FTSParams>;
+      };
+    }
+  | {
+      /**
+       * Whether the upserted vectors are of type f16 or f32.
+       *
+       * To use f16 vectors, this field needs to be explicitly specified in
+       * the schema when first creating (i.e. upserting to) a namespace.
+       */
+      vector?: {
```

Comment:
I'm not sure how useful this is, it seems to allow this in the schema definition:

```
      vector: {
        type: "[]string",
      },
```

because it matches the first variant in the union

Reply:
hmm yeah i tried so many variations to do a "if there's a vector, enforce this" kind of thing but they'd all error in some way.. let me look at this some more. There must be a way

Reply:
I'm not sure it really matters. I'd rather keep the client code simpler and let the server have the final say.

Reply:
makes sense! just got bit in the behind by the cmek test error too because i was overly specific; good reminder again not to make things overly strict here